src/librawspeed/decoders/Cr2Decoder.cpp

```diff
@@ -273,11 +273,10 @@ RawImage Cr2Decoder::decodeNewFormat() {
 
 RawImage Cr2Decoder::decodeRawInternal() {
   try {
-    if (hints.find("old_format") != hints.end()) {
+    if (hints.find("old_format") != hints.end())
       return decodeOldFormat();
-    } else {
-      return decodeNewFormat();
-    }
+
+    return decodeNewFormat();
```

Comment:
> If you are calling that code of mine "spaghetti-like" because there are more {} than necessary: please note that I merely used them here because is seemed to be the preferred style in the rest of the code base. I personally prefer to have them removed (and since I noticed, you are also not strictly following the existing example, I started to move back to my own preference ;))

You are talking about cases like
```c++
if(a)
one_line_statement();
```
?

If they fit on _ONE_ line, i think no `{}` is needed.
(see `readability-braces-around-statements.ShortStatementLines` in `.clang-tidy`)

> Nice. Even though: please don't bother changing that line here. My next pr will simply remove that whole block of code anyway :).

That is the proper solution indeed :)
I think many of these cases exist because the code can be rewritten in some more direct way that does not involve such `else after return` in the first place.

> In general, I also think it is better to discourage it. In the Huffman code below I thought it would be more understandable the way I wrote it but is admittedly a border line situation.

**RIGHT NOW** i'm not seeing too much questionable places, if there happens to be more places where it *really makes sense*, let's just disable that check.